src/cohortManager.js

```diff
@@ -0,0 +1,89 @@
+const Cohort = require('../src/cohort')
+const Student = require('../src/students')
+
+class CohortManager {
+  constructor() {
+    this.cohorts = []
+    this.previousStudent = -1
+  }
+
+  createCohort(name) {
+    const cohort = new Cohort(name)
+    this.cohorts.push(cohort)
+    return cohort
+  }
+
+  getCohortByName(name) {
+    for (let i = 0; i < this.cohorts.length; i++) {
+      if (this.cohorts[i].name === name) {
+        return this.cohorts[i]
+      }
+    }
+    return null
+  }
+
+  getCohort(name) {
+    const index = this.cohorts.findIndex((cohort) => cohort.name === name)
+    if (index === -1) {
```

Comment:
Not sure what this is looking for? You could maybe try to find by name rather than index?
if (!givenName) { return 'cohort not found' }